### PR TITLE
Add Product Recipes Template

### PR DESCRIPTION
**Pull request by tapaswenipathak**
As I am GSoD'19 applicant, prev
- https://github.com/mattermost/mattermost-api-reference/pull/448
- https://github.com/mattermost/docs/pull/2697
- https://github.com/mattermost/docs/pull/2698
- https://github.com/mattermost/docs/pull/2732

### CODE REVIEW

File: source/administration/product-receipe-template.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+Product Recipe template
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
Product Recipe Template
```

File: source/administration/product-receipe-template.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+========================
+
+This template defined how a product recipe can be added for the depending
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
This template defines how a product recipe can be added for the depending
```

File: source/administration/product-receipe-template.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+people.
+
+Problem Statement
+-------------------
+
+Define the problem statement. The problem statement could be anything from adding
+an architecture setup or installations.
+
+The problem statement should be written from user perspective and from organisation
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
The problem statement should be written from a user perspective and from an organization
```

File: source/administration/product-receipe-template.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+perspective.
+
+The user perspective might include
+
+- The requirements of the user
+- The environment of the user
+- The dependecies the user is having
+- The user case user would run the installation in
+
+The organisation perspective might include
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
The organisation perspective might include:
```

File: source/administration/product-receipe-template.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+Problem Statement
+-------------------
+
+Define the problem statement. The problem statement could be anything from adding
+an architecture setup or installations.
+
+The problem statement should be written from user perspective and from organisation
+perspective.
+
+The user perspective might include
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
The user perspective might include:
```

File: source/administration/product-receipe-template.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+- The requirements of the user
+- The environment of the user
+- The dependecies the user is having
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- The dependencies the user is having
```

File: source/administration/product-receipe-template.rst [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+- The user case user would run the installation in
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- The use case the user would run the installation in
```

File: source/administration/product-receipe-template.rst [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+The organisation perspective might include
+
+- How much effort the organisations adds in implementing the solution
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- How much effort the organization adds in implementing the solution
```

File: source/administration/product-receipe-template.rst [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+- How many work hours are put on implementing the solution for the user
+- Listing team putting the solution for the user
+- Dependecies with ongoing installations
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- Dependencies with ongoing installations
```

File: source/administration/product-receipe-template.rst [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+- The requirements of the user
+- The environment of the user
+- The dependecies the user is having
+- The user case user would run the installation in
+
+The organisation perspective might include
+
+- How much effort the organisations adds in implementing the solution
+- How many work hours are put on implementing the solution for the user
+- Listing team putting the solution for the user
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]: Not sure what this means

File: source/administration/product-receipe-template.rst [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+- Dependecies with ongoing installations
+
+Pre-installation/pre-configuration notes
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
Pre-Installation/Pre-Configuration Notes
```

File: source/administration/product-receipe-template.rst [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+----------------------------------------
+
+Define the pre-installation information. Projects installation has pre-installation
+steps describing the things you should setup when installating a requirement.
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]: "Projects installation has pre-installation
steps describing the things you should setup when installating a requirement."

Would you be open to clarifying this sentence?

File: source/administration/product-receipe-template.rst [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+Breifly describe the pre-installation or pre-configuration notes. Things you
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
Briefly describe the pre-installation or pre-configuration notes. Things you
```

File: source/administration/product-receipe-template.rst [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+might include
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
might include:
```

File: source/administration/product-receipe-template.rst [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+- Environment information
+- Version information
+- Release information
+- Dependecy information
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- Dependency information
```

File: source/administration/product-receipe-template.rst [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+- Rollback information
+- Line of Business information
+
+Instructions Setup and Configuring the Solution
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]: Would you be open to clarifying the heading?

File: source/administration/product-receipe-template.rst [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+-----------------------------------------------
+
+Define the instructions setup instructions. The instructions should be detailed
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]: Please clarify the opening sentence.

File: source/administration/product-receipe-template.rst [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+and have everything about setting up and configuring the solution. Things you
+might include
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
might include:
```

File: source/administration/product-receipe-template.rst [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+
+- Installation instructions
+- Trouble shooting instructions
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
- Troubleshooting instructions
```

File: source/administration/product-receipe-template.rst [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+- Testing the installation
+- Testing the installation in multiple environments
+
+Defining Recipe Category For Search and Retrival
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]:
```suggestion
Defining Recipe Category For Search and Retrieval
```
?

File: source/administration/product-receipe-template.rst [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+------------------------------------------------
+
+Define the categories and tag for the users. The users should be easily search
+and retrieve the receipe. Add tech, line of business, architecture, environement
+information.
```
Comment (amyblais) [inline source/administration/product-receipe-template.rst]: Please help make this paragraph a little clearer.

**Comment by amyblais** [comment]
Hi @tapaswenipathak, thank you for the PR! Some things to consider:

1. Please help review your work especially for grammar, readability and formatting.
2. Please help edit the URL to correct any typos: product-receipe-template.rst
3. Please help add any new documents to the table of contents https://docs.mattermost.com/guides/administrator.html.

Thanks,

File: source/administration/product-receipe-template.rst (hunk 10 above)
Comment (tapaswenipathak) [reply source/administration/product-receipe-template.rst]: Teams involved in enabling the solution for the users? Let's say if an installation is happening, how many levels of escalation can be present.

File: source/administration/product-receipe-template.rst (hunk 16 above)
Comment (tapaswenipathak) [reply source/administration/product-receipe-template.rst]: The actual instructions and how the solution can be implemented.

File: source/administration/product-receipe-template.rst (hunk 21 above)
Comment (tapaswenipathak) [reply source/administration/product-receipe-template.rst]: If you can tell me what tags you have in your organisation for this sort of implementation, I can include the details. I added few generic things in these lines that could let any member of mattermost org implement a product recipe and add proper tags in the documentation.

File: source/administration/product-receipe-template.rst (hunk 16 above)
Comment (tapaswenipathak) [reply source/administration/product-receipe-template.rst]: There is a ticker for this, I picked these works from the ticket.

File: source/administration/product-recipes-template.rst [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+- How much effort the organization adds in implementing the solution
+- How many work hours are put on implementing the solution for the user
+- Listing team putting the solution for the user
+- Dependencies with ongoing installations
+
+Pre-Installation/Pre-Configuration Notes
+----------------------------------------
+
+Define the pre-installation information. Breifly describe the pre-installation
+or pre-configuration notes. Things you might include
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
or pre-configuration notes. Things you might include:
```

File: source/administration/product-recipes-template.rst [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+
+- How much effort the organization adds in implementing the solution
+- How many work hours are put on implementing the solution for the user
+- Listing team putting the solution for the user
+- Dependencies with ongoing installations
+
+Pre-Installation/Pre-Configuration Notes
+----------------------------------------
+
+Define the pre-installation information. Breifly describe the pre-installation
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
Define the pre-installation information. Briefly describe the pre-installation
```

File: source/administration/product-recipes-template.rst [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+or pre-configuration notes. Things you might include
+
+- Environment information
+- Version information
+- Release information
+- Dependency information
+- Rollback information
+- Line of Business information
+
+Instructions Setup and Configuring the Solution
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
Instructions for Setting Up and Configuring the Solution
```

File: source/administration/product-recipes-template.rst [hunk 25]
```diff
@@ -0,0 +1,60 @@
...
+-----------------------------------------------
+
+Define the setup instructions for the implementation that you are running.
+The instructions should be detailed and have everything about setting up and
+implementing the solution. Things you might include
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
implementing the solution. Things you might include:
```

File: source/administration/product-recipes-template.rst [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+
+- Installation instructions
+- Troubleshooting instructions
+- Testing the installation
+- Testing the installation in multiple environments
+
+Defining Recipe Category For Search and Retrieval
+------------------------------------------------
+
+Define the categories and tag for the users. For enabling the users of product
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
Define the categories and tags for the users. For enabling the users of product
```

File: source/administration/product-recipes-template.rst [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+recipes easily search and retrieve the recipe. Add tech, line of business,
+architecture, environement information.
```
Comment (amyblais) [inline source/administration/product-recipes-template.rst]:
```suggestion
architecture and environment information.
```

File: source/administration/product-recipes-template.rst (hunk 26 above)
Comment (amyblais) [inline source/administration/product-recipes-template.rst]: "For enabling the users of product recipes easily search and retrieve the recipe."

This sentence is a little unclear and could also be combined with the first sentence.

**Comment by amyblais** [comment]
@wiersgallak This seems to be related to https://github.com/mattermost/docs/issues/2718. Would this be more of an internal process and would be added to https://docs.mattermost.com/guides/core.html? Just confirming,

**Comment by wiersgallak** [comment]
I agree.  I think that the template is more of an internal tool.  I would recommend it living under our Documentation guidelines: https://docs.mattermost.com/process/documentation-guidelines.html

File: source/process/product-recipes-template.rst [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+- Release information
+- Dependency information
+- Rollback information
+- Line of Business information
+
+Instructions for Setting Up and Configuring the Solution
+-----------------------------------------------
+
+Define the setup instructions for the implementation that you are running.
+The instructions should be detailed and have everything about setting up and
```
Comment (amyblais) [inline source/process/product-recipes-template.rst]:
```suggestion
The instructions should be detailed and include everything about setting up and
```

File: source/process/product-recipes-template.rst [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+
+- Installation instructions
+- Troubleshooting instructions
+- Testing the installation
+- Testing the installation in multiple environments
+
+Defining Recipe Category For Search and Retrieval
+------------------------------------------------
+
+Define the categories and tag for the users; enabling the users of product
```
Comment (amyblais) [inline source/process/product-recipes-template.rst]:
```suggestion
Define the categories and tag for the users, enabling the users of product
```

File: source/process/product-recipes-template.rst [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+recipes easily search and retrieve the recipe. Add tech, line of business,
```
Comment (amyblais) [inline source/process/product-recipes-template.rst]:
```suggestion
recipes to easily search and retrieve the recipe. Add tech, line of business,
```

**Comment by jasonblais** [comment]
I've added a `Lifecycle/frozen` [label](https://developers.mattermost.com/contribute/getting-started/labels/) to keep the PR open while @wiersgallak has it in her queue to review :)

**Comment by tapaswenipathak** [comment]
folks, gentle reminder for review.

File: source/process/product-recipes-template.rst [hunk 31]
```diff
@@ -0,0 +1,60 @@
+Product Recipe Template
+========================
+
+This template defines how a product recipe can be added for the depending
```
Comment (wiersgallak) [inline source/process/product-recipes-template.rst]: Consider edits to this section "for the depending people" is confusing to me.  I did find this existing documentation as well: https://docs.mattermost.com/process/guest-article.html#mattermost-recipes. We should try to incorporate this information into this template.

Comment (tapaswenipathak) [reply source/process/product-recipes-template.rst]: What do you suggest for https://github.com/mattermost/docs/issues/2718?

**Comment by wiersgallak** [comment]
Per discussion, closing this PR to be worked on in closer collaboration as a larger work effort with the team.

**Pull request closed without merging**